lib/route/link/can.c

```diff
@@ -166,6 +168,11 @@ static int can_parse(struct rtnl_link *link, struct nlattr *data,
 		ci->ci_mask |= CAN_HAS_DATA_BITTIMING_CONST;
 	}
 
+	if (xstats) {
+		nla_memcpy(&ci->ci_device_stats, xstats, sizeof(ci->ci_device_stats));
```

Comment:
`nla_memcpy()` will take care to copy only `_NL_MIN(count, nla_len(src))` bytes.

But it seems wrong to accept a truncated struct. I think it should be `if (xstats && nla_len(xstats) >= sizeof(ci->ci_devices_stats))`.